dashboard/tasks_test.py

```diff
@@ -44,26 +43,96 @@ def setUpTestData(cls):
             mocked_refresh_cache.assert_any_call(user, mock.ANY, cache_type)
             mocked_refresh.assert_any_call(user.social_auth.get(provider=EdxOrgOAuth2.name))
+
+    @mock.patch('dashboard.api_edx_cache.CachedEdxDataApi.update_cache_if_expired', new_callable=mock.MagicMock)
+    @mock.patch('backends.utils.refresh_user_token', autospec=True)
+    def test_subtask_user_does_not_exist(self, mocked_refresh, mocked_refresh_cache):
+        """
+        Test if the user has been deleted between the select and the run of the task,
+        the task still completes.
+        """
+        with mock.patch('django.contrib.auth.models.User.objects.get', autospec=True, side_effect=User.DoesNotExist):
+            batch_update_user_data_subtasks.s(self.students).apply(args=()).get()
+        assert mocked_refresh.called is False
+        assert mocked_refresh_cache.called is False
+
+    @mock.patch('dashboard.api_edx_cache.CachedEdxDataApi.update_cache_if_expired', new_callable=mock.MagicMock)
+    @mock.patch('backends.utils.refresh_user_token', autospec=True)
+    def test_subtask_user_no_social_auth(self, mocked_refresh, mocked_refresh_cache):
+        """
+        Test if the user has not a social auth the task still completes
+        """
+        user_no_social_auth = UserFactory.create()
+        batch_update_user_data_subtasks.s([user_no_social_auth]).apply(args=()).get()
+        assert mocked_refresh.called is False
+        assert mocked_refresh_cache.called is False
+
+    @mock.patch('dashboard.api_edx_cache.CachedEdxDataApi.update_cache_if_expired', new_callable=mock.MagicMock)
+    @mock.patch('backends.utils.refresh_user_token', autospec=True)
+    def test_subtask_user_wrong_social_auth(self, mocked_refresh, mocked_refresh_cache):
+        """
+        Test if the user has a wrong social auth the task still completes
+        """
+        user_wrong_social_auth = UserFactory.create()
+        user_wrong_social_auth.social_auth.create(
+            provider='foo_provider',
+            uid="{}_edx".format(user_wrong_social_auth.username),
+            extra_data={"access_token": "fooooootoken"}
+        )
+        batch_update_user_data_subtasks.s([user_wrong_social_auth]).apply(args=()).get()
+        assert mocked_refresh.called is False
+        assert mocked_refresh_cache.called is False
+
+    @mock.patch('dashboard.api_edx_cache.CachedEdxDataApi.update_cache_if_expired', new_callable=mock.MagicMock)
+    @mock.patch('backends.utils.refresh_user_token', autospec=True)
+    def test_subtask_social_auth_refresh_fails(self, mocked_refresh, mocked_refresh_cache):
+        """
+        Test if the refresh for the user's OAUTH token fails the task still completes
+        """
+        mocked_refresh.side_effect = InvalidCredentialStored
+        batch_update_user_data_subtasks.s(self.students).apply(args=()).get()
+        assert mocked_refresh.called is True
```

Comment:
Can you assert the arguments passed here and in the other cases below? We should make sure it's called once per user, or for `mocked_refresh_cache` once per user per cache type